keps/sig-storage/1710-selinux-relabeling/README.md

```diff
@@ -508,10 +542,24 @@ _This section must be completed when targeting beta graduation to a release._
 
 * **What are the SLIs (Service Level Indicators) an operator can use to
   determine the health of the service?**
+
   - [ ] Metrics
-    - Metric name:
-    - [Optional] Aggregation method:
-    - Components exposing the metric:
+    - All `errors_total` metrics below cover real errors when a Pod can't start.
+      It applies to `ReadWriteOncePod` volumes.
+    - All `warnings_total` metrics below cover **future** errors that would appear if this feature was extended to all volumes.
+      This will be evaluated in Phase 2.
+    - 1. `volume_manager_selinux_container_errors_total` + `volume_manager_selinux_container_warnings_total`: Number of errors when kubelet cannot compute SELinux context for a container.
+        This indicates an error converting SELinux context into SELinux label by github.com/opencontainers/selinux/go-selinux library.
+        Reading its source code, this should never happen, but one never knows.
+      1. `volume_manager_selinux_pod_context_mismatch_errors_total` + `volume_manager_selinux_pod_context_mismatch_warnings_total`: Number of errors when a Pod defines different SELinux contexts for its containers that use the same volume.
+         Before this feature, only one container in such a Pod could access the volume.
+         With this feature, the Pod won't even start.
+         This metric captures nr. of failed Pod starts, including periodic retries.
+      1. `volume_manager_selinux_volume_context_mismatch_errors_total` + `volume_manager_selinux_volume_context_mismatch_warnings_total`: Number of errors when a Pod uses a volume that is already mounted with a different SELinux context than the Pod needs.
+         Before this feature, both pods would start, but only one such pod could access the volume.
+         With this feature, one of the Pods won't even start.
```

Comment:
We briefly discussed rejecting such pods via pod admission in future. Are we still planning to do that?

Reply:
Kubelet admission already rejects RWOP pods that use a volume that is already used.

We could add the same for Pods with mismatching SELinux contexts, however, some volume types (e.g. NFS) might support a volume mounted on a node several times with different contexts.